test/rummager_test.rb

```diff
@@ -138,6 +138,39 @@
     assert_requested :get, /.*/, headers: { "authorization" => "token" }
   end
 
+  # tests for search_enum
+  it "#search_enum returns two pages - last page is half full" do
+    stub_request(:get, /example.com\/search/)
+      .with(query: hash_including(start: '0', count: '2'))
+      .to_return(body: { 'results' => [{ 'title' => 't1' }, { 'title' => 't2' }] }.to_json)
+
+    stub_request(:get, /example.com\/search/)
+      .with(query: hash_including(start: '2', count: '2'))
+      .to_return(body: { 'results' => [{ 'title' => 't3' }] }.to_json)
+
+    search_results = [{ 'title' => 't1' }, { 'title' => 't2' }, { 'title' => 't3' }]
+    results = GdsApi::Rummager.new("http://example.com").search_enum({}, page_size: 2)
+    assert_equal search_results, results.to_a
+  end
+
+  it "#search_enum returns two pages - last page is full" do
+    stub_request(:get, /example.com\/search/)
+      .with(query: hash_including(start: '0', count: '2'))
+      .to_return(body: { 'results' => [{ 'title' => 't1' }, { 'title' => 't2' }] }.to_json)
+
+    stub_request(:get, /example.com\/search/)
+      .with(query: hash_including(start: '2', count: '2'))
+      .to_return(body: { 'results' => [{ 'title' => 't3' }, { 'title' => 't4' }] }.to_json)
+
+    stub_request(:get, 'http://example.com/search.json')
```

Comment:
The other stubs are for `/example.com\/search/`. Could that be used here too? (for conistency...)